src/runner.js

```diff
@@ -491,11 +558,13 @@ async function run(config, testCases) {
     external_locking.prepare(config);
     const initData = config.beforeAllTests ? await config.beforeAllTests(config) : undefined;
 
-    const tasks = await testCases2tasks(config, testCases);
+    const resultById = new Map();
+    const tasks = await testCases2tasks(config, testCases, resultById);
     /** @type {RunnerState} */
     const state = {
         flakyCounts: new Map(),
         tasks,
+        resultById,
```

Comment:
From this naming, it's not clear we are using the _task_ ID here. It's a bit longer, but ca we rename it to `resultByTaskId`?

Reply:
done 👍